### PR TITLE
Handle case if no files was supplied

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Description
https://github.com/owncloud/core/pull/39198 introduced a bug, where the root dir folder download -> download.zip was empty.
Fixed with this PR

## Related Issue

https://central.owncloud.org/t/files-are-downloaded-as-a-file-without-extension/33798/13

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: lib/private/legacy/files.php [hunk 1]
```diff
@@ -111,7 +111,11 @@ public static function get($dir, $files, $params = null) {
 		}
 
 		try {
-			if (\count($listOfFiles) === 1) {
+			if(\count($listOfFiles) === 0){
```
Comment (phil-davis) [inline lib/private/legacy/files.php]:
```suggestion
			if (\count($listOfFiles) === 0){
```

**Review by phil-davis — COMMENTED** [review]
Fails coding standard. But the fix works - I select all files/folders in the root of a user and `download/zip` actually has the files/folders in it.

**Review by phil-davis — APPROVED** [review]

File: lib/private/legacy/files.php [hunk 2]
```diff
@@ -111,7 +111,10 @@ public static function get($dir, $files, $params = null) {
 		}
 
 		try {
-			if (\count($listOfFiles) === 1) {
+			if (\count($listOfFiles) === 0) {
```
Comment (JammingBen) [inline lib/private/legacy/files.php:114]: Maybe add a comment on why `$listOfFile` could be empty and why we're handling this?

**Review by JammingBen — COMMENTED** [review]
Code itself LGMT

File: lib/private/legacy/files.php [hunk 3]
```diff
@@ -111,7 +111,10 @@ public static function get($dir, $files, $params = null) {
 		}
```
Comment (phil-davis) [inline lib/private/legacy/files.php:111]: The indent displayed by GitHub is strange. This `}` ends the else of the if statement. But it is 1 tab less.

If you are touching this PR again, then maybe make a separate re-formatting commit too.

File: lib/private/legacy/files.php [hunk 4]
```diff
@@ -101,7 +101,6 @@ private static function sendHeaders($filename, $name, array $rangeArray) {
 	 */
 	public static function get($dir, $files, $params = null) {
 		$view = \OC\Files\Filesystem::getView();
-		$getType = self::FILE;
```
Comment (JanAckermann) [inline lib/private/legacy/files.php:104]: Note, that this was even not used before this PR

File: lib/private/legacy/files.php (hunk 3 above)
Comment (JanAckermann) [reply lib/private/legacy/files.php:110]: this is kind of weird

Comment (phil-davis) [reply lib/private/legacy/files.php:110]: Not important.

**Pull request merged**